dashboard/design/survey_summary_design.md

```diff
@@ -0,0 +1,103 @@
+Specifically, we need to support at least 3 new surveys in 2019, CSF Deep Dive pre & post workshop surveys and CSF Intro survey.
+
+## 1. Update current pipeline to support 2019 surveys
+We will reuse the current pipeline as much as possible to support 3 new CSF surveys in 2019.
+
+This option is based on `local_workshop_daily_survey_report` pipeline. We reuse its data model `WorkshopDailySurvey` and UI view `local_summer_workshop_daily_survey/results.jsx`. We will modify its logic in to retrieve and summarize survey results, which is currently tightly coupled to summer workshop context. This logic lives in `WorkshopSurveyReportController` and `WorkshopSurveyResultsHelper`.
+
+2 approaches:
+- Modify this logic to make it more generic with the risk of introducing regression (the code is complex).
+- Create a new path separate from the existing one, which means adding substantial duplicated code.
+
+**Pros:** Risk is small (if creating a new path). Amount of new code could is not small but manageable. We do just enough to support 3 new surveys in 2019.
+
+**Cons:** Not scalable when we have more surveys. Could become thrown-away work if we decide to move to generic pipeline later.
+
+**Cost Estimate:** 5 ± 1 days.
+
+
+## 2. Create survey report in JotForm
+JotForm support creating [Visual Report](https://www.jotform.com/help/187-How-to-Create-a-Visual-Report-with-Your-Form-Submissions) ([example](https://www.jotform.com/report/91067837377065)). We can then embed this report as an iframe into workshop dashboard. Authentication and authorization before showing report will be handled by our side.
+
+**Pros:** The cheapest option in term of engineering cost to enable. Survey owners can create report themselves. Report is automatically updated (don't need to download data locally and upload to Tableau).
+
+**Cons:** Can only do basic calculation and visualization (not the same level as Tableau or PowerBI). User can not interact with the report. Cannot pass parameter to personalize the report.
+
+**Cost Estimate:** 3 ± 1 days.
+
+## 3. Build generic survey pipeline
+### Overview
+Creating a generic pipeline that starts from ingesting survey data, making data searchable, summarizing data, to presenting result in our site.\
+Key features of the new pipeline is the decoupling of code from specific survey or workshop context, and modular design that allows more flexibility to adapt to future requirements.
+
+**Pros:** Minimize work to support new survey. Open for extensions, flexible to change such as adding/removing/updating components because of modular design. Enable new scenarios such as A/B testing in Survey.
+
+**Cons:** Expensive. (This could be mitigated by building the pipeline from the end first as discussed below.)
+
+**Cost Estimate:** 3-4 weeks for the complete pipeline. 7 ± 2 days to implement step 3 (Summarize) and the first component of step 4 (Present) first, enough to support CSF 2019 surveys.
```

Comment:
Can you provide a more detailed breakdown of this estimate, or at least the step 3 part?  I'd like to see it broken into tasks of 3 days or less.  I'd expect time for the Retriever/Modifier/Transformer/Mapper/Reducer implementations, but also time for the adapters you mentioned, buffer for tests and pipeline delays, potentially time for migrating existing survey data and validation afterward, and maybe some communication or training with team members authoring surveys so that they understand our new workflow.

Reply:
Added rollout plan and specific things we will build in Recommendation section